src/client/pfs.go

```diff
@@ -468,9 +469,10 @@ func (c APIClient) PutFileURL(repoName string, commitID string, path string, url
 		}
 	}()
 	if err := putFileClient.Send(&pfs.PutFileRequest{
-		File:     NewFile(repoName, commitID, path),
-		FileType: pfs.FileType_FILE_TYPE_REGULAR,
-		Url:      url,
+		File:      NewFile(repoName, commitID, path),
+		FileType:  pfs.FileType_FILE_TYPE_REGULAR,
+		Url:       url,
+		Recursive: recursive,
```

Comment:
It's not useful, in fact it just errors. I think that's reasonable though, the alternative would be to make it detect that it's a directory and automatically do the recursive traversal. That's a little too magical imo though, recursive puts are a different and expensive enough thing that users should have to ask for them explicitly. This is the way that most unix tools work too. For example `cp dir somewhere` will error unless you add the `-r` flag to do a recursive `cp`.